user/super/com/google/gwt/emul/java/util/stream/Collectors.java

```diff
@@ -376,12 +376,31 @@ public static <T> Collector<T,?,List<T>> toList() {
   public static <T, K, U> Collector<T, ?, Map<K, U>> toMap(
       final Function<? super T, ? extends K> keyMapper,
       final Function<? super T, ? extends U> valueMapper) {
-    return toMap(
-        keyMapper,
-        valueMapper,
+    return Collector.of(
```

Comment:
Can these two toMap methods that call Collector.of() be refactored to share common lambdas? lambdas and method references are pretty expensive to generate, and while the new error message will be shorter, the total generated code is probably going to increase for this change

Reply:
The only way I was able to reduce the number of lambdas was parametrizing the private `toMap` 🤔
The extra parameter could be avoided if we say that `mergeFunction == null` should be interpreted as `unique == true` in the private method (and possibly add a null check to the public one).

I don't think this can be easily solved with composition since either `collisionCheck` would have to return both `T` and `K` wrapped in an `Entry` or `keyMapper` would have to be called twice (in collision checking and in accumulator).